### PR TITLE
Optimize

### PR DESCRIPTION
**Pull request by raphaelspies**
All API methods now check local storage for data before calling API

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -64,25 +64,53 @@ const App = () => {
 
   // API calls
   const getProducts = async () => {
-    const fetchedProducts = await axios.get('/products');
-    return fetchedProducts.data;
+    const cachedProducts = localStorage.getItem('products');
```
Comment (erikeh) [inline client/src/components/App.jsx:67]: const cachedProducts = await localStorage.getItem('products') ?

Comment (erikeh) [reply client/src/components/App.jsx:67]: Not sure how async the getItem function is, but assuming it is async

**Review by erikeh — APPROVED** [review]
sick that we got caching implemented! just left a comment wondering whether we should await the cached data - in the off chance that a getCachedData call might take longer resulting in an error

**Review by Lordnickim — APPROVED** [review]

File: client/src/components/App.jsx (hunk 1 above)
Comment (raphaelspies) [reply client/src/components/App.jsx:67]: No, it seems to be synchronous. Works without specifying that it's async, and stackOverflow seems to think it's synchronous

Comment (raphaelspies) [reply client/src/components/App.jsx:67]: errata: there's something else call `asyncLocalStorage()` which is apparently faster: https://dev.to/createnextapp/why-asynclocalstorage-instead-of-localstorage-1ha7
Will look into it

**Pull request merged**